### PR TITLE
fix(learn): tests of the for in issue regarding the curly braces

### LINKED ISSUE #40277: Test a For loop doesn't work using short syntax.
**Describe your problem and how to reproduce it:**
Test a For loop doesn't work using short syntax.
If we got this code:
for(let user in usersObj) usersObj[user].online ? ++count : null;
The tests fail, but it works if we use the brackets syntax for the loop:
for(let user in usersObj){usersObj[user].online ? ++count : null};

**Add a Link to the page with the problem:**
https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/basic-data-structures/iterate-through-the-keys-of-an-object-with-a-for---in-statement

**Tell us about your browser and operating system:**
* Browser Name: Chrome
* Browser Version:  87
* Operating System: Windows 10

forin

### PR DESCRIPTION
**Pull request by Ismailtlem**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #40277

Just removed the curly brace from the regex expression of the first test case as it seems it's what caused the probelm

### CODE REVIEW

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/iterate-through-the-keys-of-an-object-with-a-for...in-statement.md [hunk 1]
```diff
@@ -15,10 +15,10 @@ for (let user in users) {
...
 
 // logs:
-Alan
-Jeff
-Sarah
-Ryan
+Alan;
+Jeff;
+Sarah;
+Ryan;
```
Comment (Sky020) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/iterate-through-the-keys-of-an-object-with-a-for...in-statement.md:21]: @ojeytonwilliams I assume this has to do with the new syntax (confusing the prettier config). Can we avoid this?

Comment (Ismailtlem) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/iterate-through-the-keys-of-an-object-with-a-for...in-statement.md]: I think i fixed this

Comment (ojeytonwilliams) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/iterate-through-the-keys-of-an-object-with-a-for...in-statement.md]: > @ojeytonwilliams I assume this has to do with the new syntax (confusing the prettier config). Can we avoid this?

This isn't new, it's just how prettier handles fenced code blocks.  

We can mitigate this, but it involves identifying specific cases (like this) where prettier will break something.  Nick's taking a look at this, so we can discuss a bit further when he drops his PR.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/iterate-through-the-keys-of-an-object-with-a-for...in-statement.md [hunk 2]
```diff
@@ -48,7 +48,7 @@ The function `countOnline` should use a `for in` statement to iterate through th
 ```js
 assert(
   code.match(
-    /for\s*\(\s*(var|let|const)\s+[a-zA-Z_$]\w*\s+in\s+[a-zA-Z_$]\w*\s*\)\s*{/
+    /for\s*\(\s*(var|let|const)\s+[a-zA-Z_$]\w*\s+in\s+[a-zA-Z_$]\w*\s*\)\s*/
```
Comment (Sky020) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/iterate-through-the-keys-of-an-object-with-a-for...in-statement.md:51]:
```suggestion
    /for\s*\(\s*(var|let|const)\s+[a-zA-Z_$]\w*\s+in\s+[a-zA-Z_$]\w*\s*\)/
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
No need to check for a zero+ space after the `)`

**Review by ShaunSHamilton — APPROVED** [review]
This LGTM 👍 

Thank you, for making those changes

**Review by scissorsneedfoodtoo — APPROVED** [review]
Thanks for submitting this PR @Ismailtlem, and for your thorough review @ShaunSHamilton.

Tested locally with the existing solution and another that's similar to the one mentioned https://github.com/freeCodeCamp/freeCodeCamp/issues/40277. Everything LGTM :+1: :+1:

**Comment by RandellDawson** [comment]
Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 🎉

**Pull request merged**